test/Makefile

```diff
@@ -9,7 +9,7 @@
 ###############################################################################
 MOOSE_DIR          ?= $(shell dirname `pwd`)
 FRAMEWORK_DIR      ?= $(MOOSE_DIR)/framework
-ADDITIONAL_CPPFLAGS += -Wall -Wextra
+ADDITIONAL_CPPFLAGS += -Wall -Wextra -Werror -Wno-cast-function-type
```

Comment:
This doesn't work with older compilers (and probably isn't the right way to fix this anyway).

Reply:
That's correct. It was not meant as a fix, but to get further in build to see what other issues we bump into. So I put it in a separate patch so we could drop it from the branch easily when we have a proper fix.

The next problem I had IIRC was some linking fail in exodiff. Could have been just my machine - I did not get to investigate it further. Just for the record if somebody is going to pick up this PR and continue the work.